### DISCUSSION TITLE
DNF local plugin

### DISCUSSION THREAD

**Discussion by Athwale** [rhinstaller/anaconda#4997] [Q&A]
Hi,
Is is possible to configure Anaconda to use the dnf local plugin to speed up live iso development?
https://dnf-plugins-core.readthedocs.io/en/latest/local.html

I am using livemedia-creator which runs in a virtual machine in --no-virt mode without qemu this uses Anaconda's directory install feature. 
It is downloading the same packages every time. If caching with the local plugin is possible, it would shave off about 15 minutes of package downloads.
https://weldr.io/lorax/livemedia-creator.html

Thanks

**Comment by VladimirSlavik** [comment]
- In our kickstart tests, we use http repos and cache everything with squid. Would that work for you? Sadly, the whole works is too complicated to just point you to the right bits.
- The linked livemedia-creator docs seem to have a section specifically about this: [using a proxy with repos](https://weldr.io/lorax/livemedia-creator.html#using-a-proxy-with-repos).

**Reply by Athwale** [reply]
I probably could work. Though setting it up is probably going to be more complicated than installing the plugin.
I was hoping someone tried the plugin, figured out it works and I just missed the right link in my attempts to find the howto. :D 
Anyway I guess I will have to go play with squids.

**Reply by VladimirSlavik** [reply]
This is the script we use to start squid. It runs directly on the host. https://github.com/rhinstaller/kickstart-tests/blob/master/containers/squid.sh

The `start()` part seems to be the bits you need. I think if you configure your VM to use a bridge, it should apply to it as well. But you'll probably want to look into reversibility of the networking changes - your host would be probably your physical machine. If you can do nested virtualization, then definitely test it that way first.

There's also `stop()` that seems to undo the changes. I write seems, because we actually never run this. FWIW, our hosts are cattle VMs in openstack, I routinely shell into them, and there's nothing special about networking there at a glance.

**Reply by Athwale** [reply]
Thanks.
